### PR TITLE
[bug 1981402] remediation cooldown is not working

### PR DESCRIPTION
**Pull request by n1r1**
fixes https://bugzilla.redhat.com/show_bug.cgi?id=1981402

### CODE REVIEW

File: controllers/poisonpillremediation_controller.go [hunk 1]
```diff
@@ -147,8 +147,8 @@ func (r *PoisonPillRemediationReconciler) Reconcile(ctx context.Context, req ctr
 			//it takes some time to kubelet to update its node conditions after we re-created it
 			//let's wait some time until it does (we could also check lastHeartBeat time but this feel safer)
 			//without that, MHC might think the node is unhealthy again, and will end up with remediation loop
-			if cooldownTime.Before(now) {
-				timeToRequeue := now.Sub(cooldownTime) + time.Second
+			if cooldownTime.After(now) {
```
Comment (rgolangh) [inline controllers/poisonpillremediation_controller.go:150]: other functions under time can make this bit more readable perhaps.
time.Until(cooldownTime) for example.

File: controllers/controller_test.go [hunk 2]
```diff
@@ -148,13 +149,22 @@ var _ = Describe("ppr Controller", func() {
 			}, 5*time.Second, 250*time.Millisecond).Should(BeFalse())
 		})
 
+		It("Verify that finalizer exists for remediation cooldown", func() {
+			Consistently(func() bool {
+				pprNamespacedName := client.ObjectKey{Name: unhealthyNodeName, Namespace: pprNamespace}
+				newPpr := &poisonpillv1alpha1.PoisonPillRemediation{}
+				Expect(k8sClient.Get(context.TODO(), pprNamespacedName, newPpr)).ToNot(HaveOccurred())
```
Comment (rgolangh) [inline controllers/controller_test.go:156]: Can you replace ToNot(HaveOccured() with To(Succeed()) ?

File: controllers/poisonpillremediation_controller.go (hunk 1 above)
Comment (rgolangh) [reply controllers/poisonpillremediation_controller.go:150]: anyhow, your call, this doesn't block the PR

**Comment by rgolangh** [comment]
/lgtm
Your call on using the comments in separate PRs

File: controllers/controller_test.go [hunk 3]
```diff
@@ -17,9 +17,10 @@ import (
 )
 
 const (
-	unhealthyNodeName string = "node1"
-	peerNodeName      string = "node2"
-	pprNamespace      string = "default"
+	unhealthyNodeName   string = "node1"
+	peerNodeName        string = "node2"
+	pprNamespace        string = "default"
+	remediationCooldown        = 20 * time.Second
```
Comment (rgolangh) [inline controllers/controller_test.go:23]: the string type delcleration seems redundant on those consts and the addition of `remediationCooldown` just exposed it.

Comment (n1r1) [reply controllers/controller_test.go:23]: yep

File: controllers/controller_test.go (hunk 2 above)
Comment (n1r1) [reply controllers/controller_test.go:156]: sure

File: controllers/poisonpillremediation_controller.go (hunk 1 above)
Comment (n1r1) [reply controllers/poisonpillremediation_controller.go:150]: Until is a nice syntactic sugar. I didn't know about it.
thanks!

File: controllers/controller_test.go (hunk 2 above)
Comment (n1r1) [reply controllers/controller_test.go]: changed

File: controllers/poisonpillremediation_controller.go (hunk 1 above)
Comment (n1r1) [reply controllers/poisonpillremediation_controller.go:150]: changed

**Pull request merged**